reactor-pool-micrometer/src/main/java/reactor/pool/introspection/micrometer/DocumentedPoolMeters.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) 2022 VMware Inc. or its affiliates, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package reactor.pool.introspection.micrometer;
+
+import io.micrometer.common.docs.KeyName;
+import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.docs.DocumentedMeter;
+
+/**
+ * Meters used by {@link Micrometer} utility.
+ */
+enum DocumentedPoolMeters implements DocumentedMeter {
+
+	ACQUIRED {
+		@Override
+		public String getName() {
+			return "%s.resources.acquired";
+		}
+
+		@Override
+		public Meter.Type getType() {
+			return Meter.Type.COUNTER;
```

Comment:
I've fixed the disagreeing types, but I do need to carefully consider the points that you bring about counters vs gauges. I'm not sure how to apply such a transition here, precisely because `PoolMetrics` interface exposes a "view" into the current state of the pool for the 4 metrics: ACQUIRED, ALLOCATED, IDLE, PENDING_ACQUIRE.

These numbers can indeed grow and shrink depending on when the methods are called.

How would you go about that one?

Reply:
If they are non-monotonic (can grow and shrink), it's straightforward, they need to be a `Gauge` since `Counter`s must be monotonic.

I think this also agrees with their meaning: what is the actual value in that very moment when you call the method that provides the data.
If they would be a `Counter`, their meaning would be something like: how many time did happen since app start/last method call.